Point_set_processing_3/include/CGAL/structure_point_set.h

```diff
@@ -0,0 +1,1483 @@
+// Copyright (c) 2015 INRIA Sophia-Antipolis (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+//
+//
+// Author(s)     : Florent Lafarge, Simon Giraudot
+//
+
+#ifndef CGAL_STRUCTURE_POINT_SET_3_H
+#define CGAL_STRUCTURE_POINT_SET_3_H
+
+#include <CGAL/property_map.h>
+#include <CGAL/point_set_processing_assertions.h>
+#include <CGAL/assertions.h>
+
+#include <CGAL/centroid.h>
+
+#include <CGAL/Kd_tree.h>
+#include <CGAL/Fuzzy_sphere.h>
+#include <CGAL/Fuzzy_iso_box.h>
+#include <CGAL/Search_traits_d.h>
+
+#include <CGAL/Delaunay_triangulation_3.h>
+#include <CGAL/Triangulation_vertex_base_with_info_3.h>
+
+#include <iterator>
+#include <list>
+#include <limits>
+
+// The following lines only for MS Visual C++
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable:4244) // boost converts signed to std::size_t
+#endif
+
+
+
+namespace CGAL {
+
+/*!
+\ingroup PkgPointSetProcessing
+
+\brief A 3D point set with structure information based on a set of
+detected planes.
+
+Given a point set in 3D space along with a set of fitted planes, this
+class stores a simplified and structured version of the point
+set. Each output point is assigned to one, two or more primitives
+(depending wether it belongs to a planar section, an edge or a if it
+is a vertex). The implementation follow \cgalCite{cgal:la-srpss-13}.
+
+\tparam Traits a model of `EfficientRANSACTraits`
+
+*/
+template <typename Traits>
+class Point_set_with_structure
+{
+public:
+
+  /// \cond SKIP_IN_MANUAL
```

Comment:
Aie, you hide `Input_iterator` and you are using it in a documented constructor...